test/IDE/module-aliasing-code-complete.swift

```diff
@@ -0,0 +1,128 @@
+/// Test code completion with module aliasing
+/// When -module-alias <alias_name>=<real_name> is applied, code completion should show
+/// the <alias_name> as that's the name which should appear in source files including import statements,
+/// decls, expressions, etc. while getting visible decls come from the module of <real_name>.
+/// Below, XLogging is the alias and mapped to the real name AppleLogging. Note that the real name
+/// AppleLogging should not appear in the code completion results.
+///
+// RUN: %empty-directory(%t)
+// RUN: %{python} %utils/split_file.py -o %t %s
+
+// RUN: %target-swift-frontend %t/FileLogging.swift -module-name AppleLogging -module-alias XLogging=AppleLogging -emit-module -o %t/AppleLogging.swiftmodule
+
+// RUN: %target-swift-ide-test -code-completion -code-completion-token=MODULE_NAME -source-filename %t/FileLib1.swift -module-alias XLogging=AppleLogging -I %t > %t/result1.txt
+// RUN: %FileCheck %s -check-prefix CHECK1 < %t/result1.txt
+
+// CHECK1-NOT: AppleLogging
+// CHECK1: found code completion token MODULE_NAME
+// CHECK1: Begin completions
+// CHECK1: Decl[Module]/None:                  XLogging[#Module#]; name=XLogging
```

Comment:
You should use `CHECK-DAG` for the three completion line checks so that they are allowed in any order in between the `Begin completions` and the `End completions`.